### DISCUSSION TITLE
How can i make it run in pythonanywhere.com?

### DISCUSSION THREAD

**Discussion by ghost** [Tkd-Alex/Twitch-Channel-Points-Miner-v2#327] [General]

**Comment by Rakambda** [comment]
Technically it could but I personally think that it's a bit out of the scope of this project.
You probably already have plenty of apps that you can connect on an it chat an that will perform interactions.

**Reply by ghost** [reply]
> Technically it could but I personally think that it's a bit out of the scope of this project. You probably already have plenty of apps that you can connect on an it chat an that will perform interactions.

Can you tell me an app that does that? I couldn't find any

**Reply by ghost** [reply]
The idea is that once you enter the streamer's chat, if you see a message from Nightbot, you will automatically reply.

I think it would have to do something with the Twitch Miner since for some reason you enter the IRC chat of the streamer.

**Reply by Rakambda** [reply]
> for some reason you enter the IRC chat of the streamer

This is just my point of view on the project, which isn't mine, but it is branded as "channel points miner" but since its start it kinda went in all directions with the **for some reasons you connect to IRC** to farm bots points too.
It's not especially a bad idea but IMO it makes the project a big machine that does everything and it gets a bit too much, especially when you want just the original idea of "channel points".

This addition you ask is kinda going into that direction and would basically mean integrating some kind of "dumb" chat bot to the miner. 

> Can you tell me an app that does that? I couldn't find any

I don't really know that much. mIRC does it but is a GUI IRC client for windows 
I know that WeeChat (linux CLI client) can have plugins, so that can be made that way.

But maybe the easiest is to do your own little script in python in a few lines.
This way it is really adapted to your own use case and is therefore way more customized to your own needs.

You can for example use the irc lib like this project.
I didn't test it at all but just the logic of it would be something like:
```py
import time
from irc.bot import SingleServerIRCBot

class ClientIRC(SimpleIRCClient):
    def __init__(self, username, token, channel):
        self.channel = "#" + channel
        self.__active = False
        super(ClientIRC, self).__init__([("irc.chat.twitch.tv", 6667, f"oauth:{token}")], username, username)

    def on_welcome(self, client, event):
        client.join(self.channel)

    def start(self):
        self.__active = True
        self._connect()
        while self.__active:
            try:
                self.reactor.process_once(timeout=0.2)
                time.sleep(0.01)
            except Exception as e:
                pass

    def die(self, msg="Bye, cruel world!"):
        self.connection.disconnect(msg)
        self.__active = False

    def on_pubmsg(self, client, message):
        if message.get_user() == "bottoreply": # get_user probably doesn't exists
           client.send_message("Yo") # send_message is probably named differently
```

**Reply by Rakambda** [reply]
It seems to pretty much do what you want to.
Setting it up to match `*` may work?, and restrict it to only interact on Nightbot's messages

For the oauth token you can get it here: https://twitchapps.com/tmi/

```py
from benebot import Bot

def onNightBotMessage(bot, nickname, channel, args, message):
    bot.sendMessage(channel, "Yo nightbot")
    
if __name__ == "__main__":
    bot = Bot(username="your_username", password="oauth:your_oath_token", channels=(("#streamer_name", 20)))
    bot.addCommand(function=onNightBotMessage, trigger="*", nicknames=("Nightbot"), channels=None, case_sensitive=False, starts_with=False, cooldown=None)
    bot.connect()
    bot.run()
```

But at this point I guess you can directly ask on this repo how to customize it the way you want.

**Reply by Rakambda** [reply]
Again as I said above, to me it's not the difficulty that is the problem. It's just that you have a very specific need and that including that in this project is a bit out of the scope.

What if someone wants to reply on all messages, or just all messages of a user, reply with more complex specific rules?
You basically end up doing a whole project by itself, basically like what you linked.

So to me it's better off as a separate project. Maybe you're not into coding and that's understandable. Though there's always room to learn, especially when you're on a website rather aimed toward programmers. It's not really a place where solutions are all offered key in hands.

---

This works:
```py
import time

from irc.bot import SingleServerIRCBot


class ClientIRC(SingleServerIRCBot):
    def __init__(self, username, token, channel):
        self.channel = "#" + channel
        self.__active = False
        super(ClientIRC, self).__init__([("irc.chat.twitch.tv", 6667, f"oauth:{token}")], username, username)

    def on_welcome(self, client, event):
        client.join(self.channel)

    def start(self):
        self.__active = True
        self._connect()
        while self.__active:
            try:
                self.reactor.process_once(timeout=0.2)
                time.sleep(0.01)
            except Exception as e:
                pass

    def die(self, msg="Bye, cruel world!"):
        self.connection.disconnect(msg)
        self.__active = False

    def on_pubmsg(self, client, message):
        if message.type == "pubmsg":
            content = message.arguments[0]
            author = message.source.split("!")[0]
            if author == "Nightbot" and content.startswith("!join"):
                client.privmsg(message.target, "Hey")


if __name__ == "__main__":
    client = ClientIRC("your_account_username", "your_oauth_token", "streamer_name_to_join_the_chat")
    client.start()
```

![image](https://user-images.githubusercontent.com/1688389/138604342-24f3dc6e-be36-4f5f-b6ee-4944387174f4.png)

user ending with d is what would be the nightbot, and user with a your account controlled by the script.

If you wanna copy what was sent then just send `content` instead of `"Hey"`.
